### PR TITLE
case link update changes related

### PR DESCRIPTION
**Pull request by btensay**
Changes to deploy ethos, ecm-consumer and case migration tool to demo environment

### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RET-4892

### Change description ###
Deploy ethos, ecm-consumer, ccd case migration tool to demo to migrate ecm cases for Leeds case type.

### Checklist

- [ x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

## 🤖GIPPI PR SUMMARY🤖

- Added a new image policy file for demo version of et-ccd-case-migration
- Updated image and environment settings in demo-00.yaml and demo-01.yaml for et-ccd-case-migration
- Updated MIGRATION_CASETYPE in et-ccd-case-migration.yaml file
- Updated filterTags pattern in demo-image-policy.yaml for ethos/ecm-consumer
- Updated image in demo.yaml for ethos/ecm-consumer
- Added a new demo-image-policy.yaml file for ethos/repl-docmosis-service
- Updated image in demo.yaml for ethos/repl-docmosis-service

### CODE REVIEW

File: apps/et/et-ccd-case-migration/et-ccd-case-migration.yaml [hunk 1]
```diff
@@ -8,9 +8,9 @@ spec:
     job:
       suspend: true
       useInterpodAntiAffinity: true
-      image: hmctspublic.azurecr.io/et/migration:prod-e00ecc6-20240423073055 #{"$imagepolicy": "flux-system:et-ccd-case-migration"}
+      image: hmctspublic.azurecr.io/et/migration:pr-201-01ab268-20240424170425 #{"$imagepolicy": "flux-system:et-ccd-case-migration"}
```
Comment (hsjhita) [inline apps/et/et-ccd-case-migration/et-ccd-case-migration.yaml:11]: this should always be the prod version

File: apps/et/et-ccd-case-migration/demo-image-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: image.toolkit.fluxcd.io/v1beta1
```
Comment (hsjhita) [inline apps/et/et-ccd-case-migration/demo-image-policy.yaml:1]: this file needs to be added into the automation folder if it's not already there

File: apps/et/et-ccd-case-migration/et-ccd-case-migration.yaml (hunk 1 above)
Comment (hsjhita) [reply apps/et/et-ccd-case-migration/et-ccd-case-migration.yaml:11]: and also, this line either needs to be in demo.yaml or in the demo-00 and demo-01 files

Comment (btensay) [reply apps/et/et-ccd-case-migration/et-ccd-case-migration.yaml:11]: reverted

File: apps/et/et-ccd-case-migration/demo-image-policy.yaml (hunk 2 above)
Comment (btensay) [reply apps/et/et-ccd-case-migration/demo-image-policy.yaml:1]: done

File: apps/et/et-ccd-case-migration/demo-image-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+kind: ImagePolicy
+metadata:
+  name: et-ccd-case-migration
```
Comment (hsjhita) [inline apps/et/et-ccd-case-migration/demo-image-policy.yaml:4]:
```suggestion
  name: demo-et-ccd-case-migration
```

**Review by hsjhita — DISMISSED** [review]

File: apps/et/et-ccd-case-migration/image-policy.yaml [hunk 4]
```diff
@@ -4,4 +4,4 @@ metadata:
   name: et-ccd-case-migration
 spec:
   imageRepositoryRef:
-    name: et-ccd-case-migration
+    name: et-ccd-case-migration
```
Comment (LahiruOshara) [inline apps/et/et-ccd-case-migration/image-policy.yaml:7]:
```suggestion
    name: et-ccd-case-migration
    
```

**Review by LahiruOshara — APPROVED** [review]

**Review by LahiruOshara — APPROVED** [review]

**Review by LahiruOshara — DISMISSED** [review]

**Review by hsjhita — APPROVED** [review]

**Review by furnivall — APPROVED** [review]

**Review by thomast1906 — APPROVED** [review]

**Review by LahiruOshara — APPROVED** [review]

**Pull request merged**